### ISSUE TITLE
A little bug in trainer.py

### ISSUE THREAD

**Issue by WeiQijie** [wkentaro/pytorch-fcn#57]
Hello, wkentaro. Do you find that the training loss will suddenly become larger when a new epoch starts? I've noticed that and think it is because of the incorrect use of the training and evaluating mode of the model. 
In train_epoch(), line 171, when you finish validation, I think you should change the model back into training mode.

**Comment by wkentaro** [comment]
Thank you I will check it.

**Comment by nolsigan** [comment]
My pull request resolves this issue. #56

**Comment by monaj07** [comment]
I was getting unpredictable results and I just realized that the reason was this issue.
Came here to post it and saw it here!

**Comment by wkentaro** [comment]
Sorry, currently I have no GPU left to try this. Almost all GPUs at my lab are used for other experiments.
Maybe it takes more a few days to solve this.

**Comment by monaj07** [comment]
@wkentaro Thanks. I believe one only needs to set `self.model.train()` again right after `self.validate()`, as @WeiQijie remarked.

**Comment by wkentaro** [comment]
@nolsigan @monaj07 thanks! d10dd3062824195fe5cc9d196b4af4081247bb3f should fix this.
